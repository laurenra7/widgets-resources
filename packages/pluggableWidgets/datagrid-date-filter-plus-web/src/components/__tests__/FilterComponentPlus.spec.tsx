import { render } from "enzyme";
import { createElement } from "react";
import { FilterComponent } from "../FilterComponentPlus";
import ReactDOM from "react-dom";

describe("Filter component", () => {
    beforeAll(() => {
        jest.spyOn(global.Math, "random").mockReturnValue(0.123456789);

        // @ts-ignore
        jest.spyOn(ReactDOM, "createPortal").mockReturnValue((element, node) => {
            return element;
        });
    });

    it("renders correctly", () => {
        const component = render(<FilterComponent adjustable defaultFilter="equal" />);

        expect(component).toMatchSnapshot();
    });

    it("renders correctly when not adjustable by user", () => {
        const component = render(<FilterComponent adjustable={false} defaultFilter="equal" />);

        expect(component).toMatchSnapshot();
    });

    it("renders correctly with aria labels", () => {
        const component = render(
            <FilterComponent
                adjustable
                screenReaderButtonCaption="my label"
                screenReaderInputCaption="my label"
                defaultFilter="equal"
            />
        );

        expect(component).toMatchSnapshot();
    });
});
