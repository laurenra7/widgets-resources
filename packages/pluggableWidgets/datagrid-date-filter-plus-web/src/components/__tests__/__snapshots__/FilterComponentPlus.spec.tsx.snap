// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders correctly 1`] = `
<div
  class="filter-container"
  data-focusindex="0"
>
  <div
    class="filter-selector"
  >
    <div
      class="filter-selector-content"
    >
      <button
        aria-controls="undefined-filter-selectors"
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-default filter-selector-button button-icon equal"
      >
         
      </button>
    </div>
  </div>
  <span
    class="sr-only"
    id="undefined-label"
  />
  <div
    class="react-datepicker-wrapper"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        aria-labelledby="undefined-label"
        class="form-control filter-input"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    aria-controls="datepicker_0.123456789"
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Show calendar"
    class="btn btn-default btn-calendar"
  >
    <span
      class="glyphicon glyphicon-calendar"
    />
  </button>
</div>
`;

exports[`Filter component renders correctly when not adjustable by user 1`] = `
<div
  class="filter-container"
  data-focusindex="0"
>
  <span
    class="sr-only"
    id="undefined-label"
  />
  <div
    class="react-datepicker-wrapper"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        aria-labelledby="undefined-label"
        class="form-control"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    aria-controls="datepicker_0.123456789"
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Show calendar"
    class="btn btn-default btn-calendar"
  >
    <span
      class="glyphicon glyphicon-calendar"
    />
  </button>
</div>
`;

exports[`Filter component renders correctly with aria labels 1`] = `
<div
  class="filter-container"
  data-focusindex="0"
>
  <div
    class="filter-selector"
  >
    <div
      class="filter-selector-content"
    >
      <button
        aria-controls="undefined-filter-selectors"
        aria-expanded="false"
        aria-haspopup="true"
        aria-label="my label"
        class="btn btn-default filter-selector-button button-icon equal"
      >
         
      </button>
    </div>
  </div>
  <span
    class="sr-only"
    id="undefined-label"
  >
    my label
  </span>
  <div
    class="react-datepicker-wrapper"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        aria-labelledby="undefined-label"
        class="form-control filter-input"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    aria-controls="datepicker_0.123456789"
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Show calendar"
    class="btn btn-default btn-calendar"
  >
    <span
      class="glyphicon glyphicon-calendar"
    />
  </button>
</div>
`;
